llvm/lib/Target/AMDGPU/AMDGPUGISel.td

```diff
@@ -116,6 +116,10 @@ def gi_smrd_sgpr_imm :
     GIComplexOperandMatcher<s64, "selectSmrdSgprImm">,
     GIComplexPatternEquiv<SMRDSgprImm>;
 
+def gi_smrd_prefetch_imm :
+    GIComplexOperandMatcher<s64, "selectSmrdPrefetchImm">,
+    GIComplexPatternEquiv<SMRDPrefetchImm>;
```

Comment:
This looks like new globalisel support for prefetch, this should be done separately

Reply:
This is splitting the handling of prefetch from general handling of smrd selection. There wasn't a distinction between prefetch and s_load instruction so I had to create a way to distinguish between the two. It should be a NFC change. Should this still be done separately?

Reply:
Yes